src/dbt_score/formatters/json_formatter.py

```diff
@@ -0,0 +1,100 @@
+"""JSON formatter.
+
+Shape of the JSON output:
+
+{
+    "models": {
+        "model_foo": {
+            "score": 0.5,
+            "badge": "🥈",
+            "results": {
+                "rule1": {
+                    "result": "OK",
+                    "severity": null
+                    "message": null
+                },
+                "rule2": {
+                    "result": "WARN",
+                    "severity": "medium",
+                    "message": "Model lacks a description."
+                }
+            ]
+        },
+        "model_bar": {
+            "score": 0.0,
+            "badge": "🥉",
+            "results": {
+                "rule1": {
+                    "result": "ERR",
+                    "message": "Exception message"
+                }
+            }
+        }
+    },
+    "project": {
+        "score": 0.25,
+        "badge": "🥉"
+    }
+}
+"""
+
+
+import json
+from typing import Any
+
+from dbt_score.evaluation import ModelResultsType
+from dbt_score.formatters import Formatter
+from dbt_score.models import Model
+from dbt_score.rule import RuleViolation
+from dbt_score.scoring import Score
+
+
+class JSONFormatter(Formatter):
+    """Formatter for JSON output."""
+
+    def __init__(self, *args: Any, **kwargs: Any):
+        """Instantiate formatter."""
+        super().__init__(*args, **kwargs)
+        self._model_results: dict[str, dict[str, Any]] = {}
+        self._project_results: dict[str, Any]
+
+    def model_evaluated(
+        self, model: Model, results: ModelResultsType, score: Score
+    ) -> None:
+        """Callback when a model has been evaluated."""
+        self._model_results[model.name] = {
```

Comment:
I think it would be valuable to add something like `above_threshold` on a `model` level. I think it's a property that is actionable; a `model` can have a certain badge + score, but still considered 'not good enough' based on the thresholds configurations.

Reply:
Good point! I added a `pass` boolean property to both models and project, PTAL!